compose/project.py

```diff
@@ -274,6 +274,9 @@ def remove_stopped(self, service_names=None, **options):
             service.remove_stopped(**options)
 
     def containers(self, service_names=None, stopped=False, one_off=False):
+        if service_names:
+            # Will raise NoSuchService if one of the names is invalid
+            self.get_services(service_names)
```

Comment:
I guess this works, I'm just a little weary that we're using this for it's side effects, instead of using an interface specifically for the purpose of validating the service names. I feel like this is going to break or make it harder to refactor later on.  I'd like to investigate our options for exposing this in a more redirect way.